action.yml

```diff
@@ -0,0 +1,89 @@
+
+inputs:
+  # extra jobs must be multi-line string, as there's no support for type: array for inputs
+  extra_jobs:
+    type: string
+    required: false
+    default: ''
+  # simple matrix will only run a single php 7.4 mysql 5.7 job instead of a full matrix
+  simple_matrix:
+    type: boolean
+    default: false
+  endtoend:
+    type: boolean
+    default: true
+  phpcoverage:
+    type: boolean
+    # modules on silverstripe account will ignore this and always run codecov
+    default: false
+  phplinting:
+    type: boolean
+    default: true
+  phpunit:
+    type: boolean
+    default: true
+  js:
+    type: boolean
+    default: true
+
+# https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions
+outputs:
+  matrix:
+    description: JSON matrix
+    value: ${{ steps.php-script.outputs.matrix }}
+
+runs:
+  using: composite
+  steps:
+
+    - name: Checkout code
+      uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2.4.2
+
+    - name: Install PHP
+      uses: shivammathur/setup-php@3eda58347216592f618bb1dff277810b6698e4ca # v2.19.1
+      with:
+        php-version: '7.4'
+        extensions: yaml
+
+    - name: Create __inputs.yml
+      shell: bash
+      # Add string inputs to memory instead of using string substituion in shell script
+      # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable
+      env:
+        EXTRA_JOBS: ${{ inputs.extra_jobs }}
+        GITHUB_REPOSITORY: ${{ github.repository }}
+        # github.base_ref is the target branch on a pull-request
+        # github.ref_name is the name of the branch on push, and the tag on tag
+        GITHUB_MY_REF: ${{ github.base_ref && github.base_ref || github.ref_name }}
+      run: |
+        # Escape double quotes '"' => '\"'
+        EXTRA_JOBS=${EXTRA_JOBS//\"/\\\"}
+        GITHUB_MY_REF=${GITHUB_MY_REF//\"/\\\"}
+        if [ -f __inputs.yml ]; then
+          rm __inputs.yml
+        fi
+        touch __inputs.yml
+        echo "endtoend: ${{ inputs.endtoend }}" >> __inputs.yml
+        echo "js: ${{ inputs.js }}" >> __inputs.yml
+        echo "phpcoverage: ${{ inputs.phpcoverage }}" >> __inputs.yml
+        echo "phplinting: ${{ inputs.phplinting }}" >> __inputs.yml
+        echo "phpunit: ${{ inputs.phpunit }}" >> __inputs.yml
+        echo "simple_matrix: ${{ inputs.simple_matrix }}" >> __inputs.yml
+        echo "github_repository: $GITHUB_REPOSITORY" >> __inputs.yml
+        echo "github_my_ref: $GITHUB_MY_REF" >> __inputs.yml
+        if [[ "$EXTRA_JOBS" != "" ]]; then
+          echo "extra_jobs:" >> __inputs.yml
+          echo "$EXTRA_JOBS" >> __inputs.yml
+        fi
+        echo "cat __inputs.yml"
+        cat __inputs.yml
+
+    - name: Run php script
+      id: php-script
+      shell: bash
+      run: |
+        MATRIX_JSON=$(php ${{ github.action_path }}/action.php)
+        echo "MATRIX_JSON: $MATRIX_JSON"
```

Comment:
The JSON here will be on a single line. Is there a simple way to pretty-print it just for this output?
If it's not simple to do don't worry about it.

Reply:
Updated

Reply:
Nothing changed?

Reply:
I have an comment in job_creator.php which saying `// output json, keep it on a single line so do not use pretty print` - I'm not completely clear on the original reason, though likely it was a limitation of the output variables within github actions (see the line below)

Regardless, the echo is really on for debugging, and it only takes an extra second or two to read on a single line

Reply:
Okay cool. The "Updated" had me thinking you'd made (or intended to make) a change that makes it pretty-print.
If it's too much effort I'm not fussed.